tests/unit/IPC/hipIpcEventHandle.cpp

```diff
@@ -0,0 +1,201 @@
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+/* HIT_START
+ * BUILD: %t %s ../test_common.cpp NVCC_OPTIONS -std=c++11
+ * TEST: %t
+ * HIT_END
+ */
+
+#include "test_common.h"
+#include <sys/mman.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <semaphore.h>
+
+#define N 10000
+#define Nbytes N*sizeof(int)
+
+int enablePeers(int dev0, int dev1) {
+    int canAccessPeer01, canAccessPeer10;
+    HIPCHECK(hipDeviceCanAccessPeer(&canAccessPeer01, dev0, dev1));
+    HIPCHECK(hipDeviceCanAccessPeer(&canAccessPeer10, dev1, dev0));
+    if (!canAccessPeer01 || !canAccessPeer10) {
+        return -1;
+    }
+
+    HIPCHECK(hipSetDevice(dev0));
+    HIPCHECK(hipDeviceEnablePeerAccess(dev1, 0 /*flags*/));
+    HIPCHECK(hipSetDevice(dev1));
+    HIPCHECK(hipDeviceEnablePeerAccess(dev0, 0 /*flags*/));
+    HIPCHECK(hipSetDevice(dev0));
+
+    return 0;
+};
+
+typedef struct mem_handle {
+    hipIpcEventHandle_t eventHandle;
+    int devCount;
+} hip_ipc_t;
+
+bool runipcTest()
+{
+    sem_t *sem_ob1, *sem_ob2;
+    
+    if ((sem_ob1 = sem_open ("/my-sem-object1", O_CREAT|O_EXCL, 0660, 0)) == SEM_FAILED) {
+        perror ("my-sem-object1");
+	exit (1);
+    }
+    if ((sem_ob2 = sem_open ("/my-sem-object2", O_CREAT|O_EXCL, 0660, 0)) == SEM_FAILED) {
+        perror ("my-sem-object2");
+	exit (1);
+    }
+
+    hip_ipc_t *shrd_mem=(hip_ipc_t *)mmap(NULL, sizeof(hip_ipc_t), PROT_READ | PROT_WRITE, MAP_SHARED | MAP_ANONYMOUS, 0, 0);
+    if (shrd_mem == MAP_FAILED) {
+        perror ("mmap failed");
+        exit (1);
+    }
+    
+    pid_t pid;
+
+    pid = fork();
+    
+    if(pid != 0) {
+    
+        int dev0 = 0;
+        int dev1 = 1;
+        HIPCHECK(hipGetDeviceCount(&shrd_mem->devCount));
+        if (shrd_mem->devCount > 1) {
+           if (enablePeers(dev0, dev1) == -1) {
+              printf("warning : could not find peer gpus, only single device test is run\n");
+              shrd_mem->devCount = 1;
+           }
+        } 
+
+        hipEvent_t start;
+        HIPCHECK(hipEventCreateWithFlags(&start, hipEventDisableTiming|hipEventInterprocess));
+
+        HIPCHECK(hipIpcGetEventHandle(&shrd_mem->eventHandle, start));
+
+        HIPCHECK(hipDeviceSynchronize());
+	
+        if(sem_post(sem_ob1) == -1) {
+            perror ("sem_post error!"); 
+            exit (1);
+        }
+ 
+    } else {
+ 		
+        if(sem_wait(sem_ob1) == -1) {
+            perror ("sem_wait error!"); 
+            exit (1);
+	}
+         
+        hipEvent_t ipc_event;
+        hipStream_t stream;
+        HIPCHECK(hipStreamCreate(&stream));
+  
+        HIPCHECK(hipIpcOpenEventHandle(&ipc_event, shrd_mem->eventHandle));
+        
+        HIPCHECK(hipEventRecord(ipc_event, NULL));
+
+        HIPCHECK(hipEventRecord(ipc_event, stream))
+        
+        HIPCHECK(hipStreamWaitEvent(stream, ipc_event, 0));
+        
+        HIPCHECK(hipEventSynchronize(ipc_event));
```

Comment:
May I know why do we need two synchronizations (hipStreamWaitEvent & hipEventSynchronize)?

Reply:
Added the description. This only verifies the validity of the ipc event object with various event APIs.